pkg/xds/sync/dataplane_proxy_builder.go

```diff
@@ -80,174 +62,98 @@ func (p *DataplaneProxyBuilder) Build(key core_model.ResourceKey, envoyContext *
-	outbound := xds_topology.BuildEndpointMap(meshContext.Resource, p.Zone, meshContext.Dataplanes.Items, zoneIngresses.Items, matchedExternalServices, p.DataSourceLoader)
+	outbound := xds_topology.BuildEndpointMap(meshContext.Resource, p.Zone, meshContext.Dataplanes.Items, meshContext.ZoneIngresses.Items, matchedExternalServices, p.DataSourceLoader)
 
 	routing := &xds.Routing{
 		TrafficRoutes:   routes,
 		OutboundTargets: outbound,
-		VipDomains:      domains,
 	}
 	return routing, destinations, nil
 }
 
-func (p *DataplaneProxyBuilder) matchPolicies(ctx context.Context, meshContext xds_context.MeshContext, dataplane *core_mesh.DataplaneResource, outboundSelectors xds.DestinationMap) (*xds.MatchedPolicies, error) {
-	healthChecks, err := xds_topology.GetHealthChecks(ctx, dataplane, outboundSelectors, p.CachingResManager)
-	if err != nil {
-		return nil, err
+func (p *DataplaneProxyBuilder) resolveVIPOutbounds(meshContext xds_context.MeshContext, dataplane *core_mesh.DataplaneResource) {
+	if dataplane.Spec.Networking.GetTransparentProxying() == nil {
+		return
 	}
-
-	circuitBreakers, err := xds_topology.GetCircuitBreakers(ctx, dataplane, outboundSelectors, p.CachingResManager)
-	if err != nil {
-		return nil, err
+	// Update the outbound of the dataplane with the generatedVips
+	generatedVips := map[string]bool{}
+	for _, ob := range meshContext.VIPOutbounds {
+		generatedVips[ob.Address] = true
 	}
-
-	trafficTrace, err := xds_topology.GetTrafficTrace(ctx, dataplane, p.CachingResManager)
-	if err != nil {
-		return nil, err
-	}
-
-	retries, err := xds_topology.GetRetries(ctx, dataplane, outboundSelectors, p.CachingResManager)
-	if err != nil {
-		return nil, err
-	}
-
-	matchedPermissions, err := p.PermissionMatcher.Match(ctx, dataplane, meshContext.Resource)
-	if err != nil {
-		return nil, err
-	}
-
-	trafficLogs, err := p.LogsMatcher.Match(ctx, dataplane)
-	if err != nil {
-		return nil, err
+	outbounds := meshContext.VIPOutbounds
+	for _, outbound := range dataplane.Spec.Networking.GetOutbound() {
+		if generatedVips[outbound.Address] { // Useful while we still have resources with computed vip outbounds
+			continue
+		}
+		outbounds = append(outbounds, outbound)
 	}
+	dataplane.Spec.Networking.Outbound = outbounds
+}
 
-	faultInjection, err := p.FaultInjectionMatcher.Match(ctx, dataplane, meshContext.Resource)
+func (p *DataplaneProxyBuilder) matchPolicies(meshContext xds_context.MeshContext, dataplane *core_mesh.DataplaneResource, outboundSelectors xds.DestinationMap) (*xds.MatchedPolicies, error) {
+	matchedPermissions, err := permissions.BuildTrafficPermissionMap(dataplane, meshContext.Resource, meshContext.TrafficPermissions().Items)
 	if err != nil {
 		return nil, err
 	}
 
-	timeouts, err := xds_topology.GetTimeouts(ctx, dataplane, p.CachingResManager)
+	faultInjection, err := faultinjections.BuildFaultInjectionMap(dataplane, meshContext.Resource, meshContext.FaultInjections().Items)
```

Comment:
Both `BuildFaultInjectionMap` and `BuildRateLimitMap` return error because of the same code:
```go
additionalInbounds, err := manager_dataplane.AdditionalInbounds(dataplane, mesh)
if err != nil {
	return nil, errors.Wrap(err, "could not fetch additional inbounds")
}
```
I think this code could be extracted and placed next to the `p.resolveVIPOutbounds(meshContext, dataplane)`. So we will add outbounds and inbounds in the same place and also we get rid of returning `error` for `BuildFaultInjectionMap` and `BuildRateLimitMap`

Reply:
I extracted AdditionalInbound to DataplaneProxyBuilder. I believe we cannot just add this as inbounds of Dataplane because it will break inbound_proxy_generator and potentially other stuff.